src/main/java/appeng/items/storage/ItemSpatialStorageCell.java

```diff
@@ -104,102 +99,80 @@ public WorldCoord getStoredSize( final ItemStack is )
 		if( is.hasTagCompound() )
 		{
 			final NBTTagCompound c = is.getTagCompound();
-			if( Platform.isServer() )
-			{
-				final int dim = c.getInteger( "StorageDim" );
-				return WorldData.instance().dimensionData().getStoredSize( dim );
-			}
-			else
-			{
-				return new WorldCoord( c.getInteger( "sizeX" ), c.getInteger( "sizeY" ), c.getInteger( "sizeZ" ) );
-			}
-		}
-		return new WorldCoord( 0, 0, 0 );
-	}
-
-	@Override
-	public WorldCoord getMin( final ItemStack is )
-	{
-		final World w = this.getWorld( is );
-		if( w != null )
-		{
-			final NBTTagCompound info = (NBTTagCompound) w.getWorldInfo().getAdditionalProperty( "storageCell" );
-			if( info != null )
-			{
-				return new WorldCoord( info.getInteger( "minX" ), info.getInteger( "minY" ), info.getInteger( "minZ" ) );
-			}
+			return new WorldCoord( c.getInteger( "sizeX" ), c.getInteger( "sizeY" ), c.getInteger( "sizeZ" ) );
 		}
 		return new WorldCoord( 0, 0, 0 );
 	}
 
 	@Override
-	public WorldCoord getMax( final ItemStack is )
+	public int getStoredDimensionID( final ItemStack is )
 	{
-		final World w = this.getWorld( is );
-		if( w != null )
+		if( is.hasTagCompound() )
 		{
-			final NBTTagCompound info = (NBTTagCompound) w.getWorldInfo().getAdditionalProperty( "storageCell" );
-			if( info != null )
-			{
-				return new WorldCoord( info.getInteger( "maxX" ), info.getInteger( "maxY" ), info.getInteger( "maxZ" ) );
-			}
+			final NBTTagCompound c = is.getTagCompound();
+			return c.getInteger( "StorageCellID" );
 		}
-		return new WorldCoord( 0, 0, 0 );
+		return -1;
 	}
 
 	@Override
-	public TransitionResult doSpatialTransition( final ItemStack is, final World w, final WorldCoord min, final WorldCoord max, final boolean doTransition )
+	public TransitionResult doSpatialTransition( final ItemStack is, final World w, final WorldCoord min, final WorldCoord max, int playerId )
 	{
-		final WorldCoord scale = this.getStoredSize( is );
-
 		final int targetX = max.x - min.x - 1;
 		final int targetY = max.y - min.y - 1;
 		final int targetZ = max.z - min.z - 1;
 		final int maxSize = this.getMaxStoredDim( is );
 
-		World destination = this.getWorld( is );
+		final BlockPos targetSize = new BlockPos( targetX, targetY, targetZ );
+
+		ISpatialDimension manager = this.getSpatialDimension();
 
-		if( ( scale.x == 0 && scale.y == 0 && scale.z == 0 ) || ( scale.x == targetX && scale.y == targetY && scale.z == targetZ ) )
+		int cellid = this.getStoredDimensionID( is );
+		if( cellid < 0 )
 		{
-			if( targetX <= maxSize && targetY <= maxSize && targetZ <= maxSize )
+			cellid = manager.createNewCellDimension( targetSize, playerId );
+		}
+
+		try
+		{
+			if( manager.isCellDimension( cellid ) )
 			{
-				if( destination == null )
+				BlockPos scale = manager.getCellContentSize( cellid );
+
+				if( scale.equals( targetSize ) )
 				{
-					destination = this.createNewWorld( is );
-				}
+					if( targetX <= maxSize && targetY <= maxSize && targetZ <= maxSize )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)